### DISCUSSION TITLE
`#[route]` macro without `method`

### DISCUSSION THREAD

**Discussion by tp971** [actix/actix-web#3259] [Ideas] [answered]
Currently, the `route` macro requires a method, but I need a way to define a route for all HTTP methods. The only way I found how to do that is with `app.route("/some/path", web::route().to(some_handler))`, but I think it would be nice to define a handler with `#[route("/some/path")]` and then just do `app.service(some_handler)`.

**Comment by robjtede** [comment, marked as answer]
```
#[route("/some/path")]
```

I can see this being dangerous to allow in the general case. You can imagine a scenario where a developer thinks this is just for GETs and then deploying an app where it's problematic to receive a POST.

I think safer to push people towards the existing solutions, either:
- specifying all the supported HTTP methods using `#[route("/", method = GET, method = POST, ...)]`;
- using manual routing where `.route("/", web::route().to(handler))` or a resource with `.default_service(handler)` configured achieves it for arbitrary methods, too.

**Reply by tp971** [reply]
I guess if that's in issue, maybe make it explicit, e.g. `#[any("...")]`, or `#[route("...", method = any)]`.

**Reply by robjtede** [reply]
That could work. I'd accept a feature request issue for it.
